get.jenkins.io.tf

```diff
@@ -0,0 +1,44 @@
+# Storage account
+resource "azurerm_resource_group" "get_jenkins_io" {
+  name     = "get-jenkins-io"
+  location = var.location
+  tags     = local.default_tags
+}
+
+resource "azurerm_storage_account" "get_jenkins_io" {
+  name                     = "getjenkinsio"
+  resource_group_name      = azurerm_resource_group.get_jenkins_io.name
+  location                 = azurerm_resource_group.get_jenkins_io.location
+  account_tier             = "Premium"
+  account_replication_type = "ZRS"
+  min_tls_version          = "TLS1_2" # default value, needed for tfsec
+
+  tags = local.default_tags
+
+  # Adding a network rule with `public_network_access_enabled` set to `true` (default) selects the option "Enabled from selected virtual networks and IP addresses"
+  network_rules {
+    default_action = "Deny"
+    ip_rules = flatten(
+      concat(
+        [for key, value in module.jenkins_infra_shared_data.admin_public_ips : value],
+      )
+    )
+    virtual_network_subnet_ids = [
+      data.azurerm_subnet.trusted_ci_jenkins_io_ephemeral_agents.id,
+      data.azurerm_subnet.trusted_ci_jenkins_io_permanent_agents.id,
+      data.azurerm_subnet.trusted_ci_jenkins_io_sponsorship_ephemeral_agents.id,
+      data.azurerm_subnet.publick8s_tier.id,
+      data.azurerm_subnet.privatek8s_tier.id,
+      data.azurerm_subnet.infra_ci_jenkins_io_sponsorship_ephemeral_agents.id,
+    ]
+    bypass = ["Metrics", "Logging", "AzureServices"]
+  }
+}
+
+
+resource "azurerm_storage_share" "get_jenkins_io" {
+  name                 = "mirrorbits"
+  storage_account_name = azurerm_storage_account.get_jenkins_io.name
+  quota                = 1024 # 512.14GiB (Begining 2024)
```

Comment:
As per https://azure.microsoft.com/en-us/pricing/details/storage/files/ , this value seems to define "how much" are we going to pay:

<img width="1671" alt="Capture d’écran 2024-01-24 à 17 50 07" src="https://github.com/jenkins-infra/azure/assets/1522731/4d5d73bf-2179-4b8d-832e-bb1960766972">

> (Premium) $0.20 per provisioned GiB*
> # ...
> The provisioned size of a premium file share is specified by quota field via the API


https://learn.microsoft.com/en-us/azure/storage/files/understanding-billing#provisioned-model explains that the provisioning can be increased dynamically at any moment (while decreasing require 24h colling period between 2 requests).

WDYT about provisioning `700` Gb instead of `1024` for today (applying the 80% rule and add ~50 GB of margin) for an expected price of 140$ per month.

```suggestion
  quota                = 700 # 512.14GiB (Begining 2024)
```